gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/resource/ApplicationsResource.java

```diff
@@ -149,18 +164,50 @@ public Response getApplications(
             List<String> idsOrderedBySubscriptions = new ArrayList<>(
                 filteringService.getApplicationsOrderByNumberOfSubscriptions(applicationIds)
             );
-            orderingComparator = Comparator.comparingInt(idsOrderedBySubscriptions::indexOf);
+            orderingComparator = Comparator.comparingInt(value -> idsOrderedBySubscriptions.indexOf(value.getId()));
         } else {
             orderingComparator = Comparator.comparing(ApplicationListItem::getName, String.CASE_INSENSITIVE_ORDER);
         }
+
+        List<ApplicationListItem> applicationsList = appSupplier.get().sorted(orderingComparator).collect(Collectors.toList());
         boolean isAsc = !order.startsWith("-");
         if (!isAsc) {
-            orderingComparator.reversed();
+            Collections.reverse(applicationsList);
         }
+        return createListResponse(applicationsList, paginationParam);
+    }
 
-        List<ApplicationListItem> applicationsList = appSupplier.get().sorted(orderingComparator).collect(Collectors.toList());
+    @Override
+    protected Map<String, Map<String, Object>> computeMetadata(
+        Map<String, Map<String, Object>> metadata,
+        Map<String, Object> dataMetadata,
+        Map<String, Object> paginationMetadata,
+        List<ApplicationListItem> pageContent
+    ) {
+        Map meta = super.computeMetadata(metadata, dataMetadata, paginationMetadata, pageContent);
+        final String userId = getAuthenticatedUser();
+        List<String> applicationIds = pageContent
+            .stream()
+            .filter(
+                app ->
+                    permissionService.hasPermission(userId, RolePermission.APPLICATION_SUBSCRIPTION, app.getId(), RolePermissionAction.READ)
```

Comment:
Is there a more efficient way to do that ?
For example, by reading only once all applications on which user has reading access. Instead of multiple reads (one per app).

Reply:
No easy to improve... We will work on this with the work on permissions